test/test_scandir.py

```diff
@@ -250,6 +250,23 @@ def test_unicode(self):
         self.assertEqual(entry.name, 'file1.txt')
         self.assertEqual(entry.path, os.path.join(path, 'file1.txt'))
 
+    def test_os_walk_unicode_handling(self):
+        if sys.platform == 'win32':
+            return self.skipTest("Not windows")
+        subdir = os.path.join(TEST_PATH, 'walk').encode('utf-8')
+        dirname_utf8 = u'\uF10F\uD20D\uB30B'.encode('utf-8')
```

Comment:
What's the point of testing a non-ASCII path? You may get errors unrelated to the test. Just use an ASCII path.